### ISSUE TITLE
Global procs are not handled correctly

### ISSUE THREAD

**Issue by BraveMole** [wixoaGit/OpenDream#515]
![image](https://user-images.githubusercontent.com/25491240/145115665-4fe0c5be-d72b-4c2f-99cc-a3b0906adf41.png)

is detected as a compilation error
![image](https://user-images.githubusercontent.com/25491240/145115783-e9c8e266-6bc5-4138-b0af-3344fd211f71.png)

Removing either of the /proc/ will prevent compiling with DM

**Comment by Farseer2** [comment]
Not very familiar with DM, so I'm not going to touch it, but it seems kind of intended from whomever wrote it:
```cpp
        public bool HasProc(string name) {
            if (Procs.ContainsKey(name)) return true; // Doesn't happen

            return Parent?.HasProc(name) ?? false; // The parent has the proc
        }
```

**Comment by wixoaGit** [comment]
This is happening because OpenDream doesn't currently support global procs.

`/proc/stack_trace(msg)` should define a global proc that isn't attached to any type, and `/datum/proc/stack_trace(msg)` should define a proc attached to `/datum`. Instead OpenDream is attaching the first proc to a `/` type that gets inherited by every other type.

**Comment by wixoaGit** [comment]
Fixed by #439.
